cpp/game/board.cpp

```diff
@@ -2639,3 +2639,57 @@ bool Board::isAdjacentToPlaHead(Player pla, Loc loc, Loc plaHead) const {
   );
   return false;
 }
+
+vector<bool> Board::markSymmetricDuplicativeLoc() {
+  vector<bool> isLocSymDuplicative(MAX_ARR_SIZE, false);
+  //this function is the same as in the nninputs.cpp getSymLoc function
```

Comment:
No, please do not duplicate the code. This entire function should also move to be within the same SymmetryHelpers namespace, and use the pre-existing getSymLoc function.

Any time you are are duplicating the entire implementation of an existing function, you should reconsider and try to find an approach that avoids that.

Reply:
And yes, the symmetries code living in nninputs.h right now is a little weird. It should probably be moved to a separate .h and .cpp file. You are welcome to leave it alone for now, for someone else to improve later, or to fix it now in this PR, but please don't make it even worse. :)